### ISSUE TITLE
Socket example bug?

### ISSUE THREAD

**Issue by arnoutdekimo** [u-blox/ubxlib#160]
Hi,

In example/sockets/main.c (and others), we find the following code:

```
sock = uSockCreate(devHandle,
                               U_SOCK_TYPE_STREAM,
                               U_SOCK_PROTOCOL_TCP);

            // Make a TCP connection to the server using
            // the socket
            if (uSockConnect(sock, &address) == 0) {
                // Send the data over the socket
                // and print the echo that comes back
                uPortLog("Sending data...\n");
                while ((x >= 0) && (txSize > 0)) {
                    x = uSockWrite(sock + (sizeof(message) - txSize),
                                   message, txSize);
                    if (x > 0) {
                        txSize -= x;
                    }
                }
```

Note in particular this line in the loop:

`uSockWrite(sock + (sizeof(message) - txSize),  message, txSize);`

Isn't this a bug? Shouldn't the offset be at the message and NOT in the socket descriptor? 

Either this, or the definition of what a socket is, is completely different from any posix socket?

Kind regards,
Arnout

**Comment by RobMeades** [comment]
Yes, it is, I spotted that myself a little while ago, just hadn't got around to fixing it yet, will do now.

**Comment by RobMeades** [comment]
Fixed in 93608d2da3df3b55496c092f987fcb503f462b56.  Gonna close this one: please feel free to re-open.
